CURSOR.md

```diff
@@ -0,0 +1,130 @@
+You are an expert in JavaScript, ReactJS, Node.js, MongoDB, mongoose, and TailwindCSS.
+
+Code Style and Structure
+- Write Clear and Functional JavaScript: Use functional programming patterns, React hooks (only useState and useEffect), and modular, reusable code structures. Avoid object-oriented programming when possible.
+- Focus on Simplicity (KIS): Break complex logic into smaller, readable pieces. Use early returns and avoid unnecessary abstractions.
+- Encourage Consistent Practices: Write code that looks authored by a single person across projects. Prioritize clarity, maintainability, and adherence to shared conventions.
+- Descriptive Naming: Use meaningful and precise names for variables and functions (e.g., fetchUsers, isLoading, handleDelete).
+- Scalable File Structures: Organize projects into predictable folders like components, scenes, services, and utils for a consistent architecture.
+- Modern Frontend Standards: Use Tailwind CSS for styling and React functional components for UI. Default to responsive design using grids and embrace modern layout principles.
+- Backend Consistency: Ensure APIs have a single responsibility per endpoint. Use flat data structures and standardize API responses in {ok, data, error} format.
+- Optimize Collaboration: Keep pull requests (PRs) small and focused to ensure seamless deployments with minimal conflicts.
+- Leverage Monorepo: Use a single repository for related codebases to improve code reuse and reduce overhead.
+- Documentation and Automation: Maintain comprehensive documentation and use automation tools for testing, deployment, and repetitive tasks.
+- Iterate Quickly During MVP Development: Prioritize speed over perfection in early stages. Hardcode non-sensitive credentials to enable rapid prototyping.
+
+Naming Conventions
+- Directories: Use lowercase with dashes for folder names (e.g., components/user-profile, services/auth-service).
+- Files: Align file names with their folders using lowercase with dashes (e.g., auth-service.js in services/auth-service).
+- Component Exports: Use named exports for components and helpers. Prefer export const UserProfile = () => {} over export default UserProfile.
+- Variables and Functions: Use camelCase for naming variables and functions (e.g., fetchData, isLoading, handleDelete).
+- Constants: Use UPPER_CASE_WITH_UNDERSCORES for constants (e.g., API_BASE_URL, DEFAULT_PAGE_SIZE).
+- React Components: Use PascalCase for component names (e.g., UserProfile, LoginButton).
+- CSS/Tailwind Utilities: Follow kebab-case for Tailwind utility classes (e.g., text-gray-700, max-w-7xl).
+- Route Parameters: Use snake_case for dynamic route parameters (e.g., /users/:user_id).
```

Comment:
i wouldn't say user_id , just :id since we are already saying user on the first part of the sentence